far/vmenu.cpp

```diff
@@ -282,6 +282,74 @@ namespace
 		return !(Item.Flags & (LIF_HIDDEN | LIF_FILTERED));
 	}
 
+	int get_item_visual_length(const bool ShowAmpersand, const string_view ItemName)
+	{
+		return static_cast<int>(ShowAmpersand ? visual_string_length(ItemName) : HiStrlen(ItemName));
+	}
+
+	enum class item_hscroll_policy
+	{
+		unbound,            // The item can move freely beyond window edges.
```

Comment:
Do I understand correctly that we need `unbound` to be able to scroll as much as needed to see long lines without losing the "centered" state if it's enabled?

It looks like currently the "Find All" menu is always unbound. This contradicts with other menus and will likely make someone unhappy because muscle memory etc.
Can we make it unbound only when needed, e.g. after Ctrl+F5 and until Alt+Home/End?

Reply:
> Do I understand correctly that we need `unbound` to be able to scroll as much as needed to see long lines without losing the "centered" state if it's enabled?

It is also possible to move the vertically aligned block of annotations totally off the viewport, but yes, we need to be able to move short items off screen to keep the alignment.

> Can we make it unbound only when needed, e.g. after Ctrl+F5 and until Alt+Home/End?

Not I just thought of it, I had a full implementation of a variant behaving exactly this way. And another one where the "en bloc" mode, as I called it, would be turned on and off explicitly by a key combination. When I played with these variants, I found them quite awkward. To start with, this behavior cries for a visual mode indicator, otherwise the user cannot know what will happen when he presses `Alt+Arrow` and sometimes gets surprised. (Once I was called to something in the middle of testing, and when I returned, I could not understand what was going on until I realized that the actual mode was not what I expected,) When I tried to mash up such mode indicator, I found that it is either clutters the UI or is not prominent enough; there seem to be no good middle ground.

Another issue with "en bloc" mode approach is with horizontal scrolling of individual items. This scrolling should behave differently in different modes. The "normal" mode is supposed to be fully compatible with the current behavior, meaning `bound_stick_to_left`. On the other hand, I need `cling_to_edge` in en bloc mode because I want to be able to manually align an individual item among its neighbors. The result looks strange and also somewhat unpredictable, especially with short items which either move freely (en bloc mode) or do not move at all ("normal") mode.

I decided that instead of having complex and sometimes unpredictable behavior, it is better to change the behavior of "Find all" menu for good, while keeping it simple, uniform and predictable. I do not think compatibility with other menus, like histories or "Find all files," is a deal breaker here because "Find all," with all its highlights and line/position numbers, already looks and feels very special anyway.

Let's try to go with the current behavior and collect users' feedback. If the reaction is very negative, my next preference would be a `far:config` flag which turns this feature on and off. If not that, I will introduce the "en bloc" mode in a follow up PR.

Reply:
Thought about it all a bit more. A `far:config` flag looks to me like a good idea. I could not decide on a good name / folder though. If you can suggest something, I will add it here.

Reply:
Ok, I guess this mode makes more sense in this particular menu at least because it doesn't break the layout spectacularly.
Let's keep it as is.